packages/ingest/src/lock.ts

```diff
@@ -31,30 +43,54 @@ async function checkOldLocks(bucket, locks = []) {
 * @param {string} providerName - The provider name
 * @returns {integer} - Number of current locks in the bucket
 **/
-async function countLock(bucket, providerName) {
+export async function countLock(
+  bucket: string,
+  providerName: string
+): Promise<number> {
   const s3Objects = await listS3ObjectsV2({
     Bucket: bucket,
     Prefix: `${lockPrefix}/${providerName}`
   });
-  return checkOldLocks(bucket, s3Objects);
+
+  if (s3Objects === undefined) return 0;
+
+  const locks = <Lock[]>s3Objects.filter(({ Key }) => Key !== undefined);
+
+  return checkOldLocks(bucket, locks);
 }
 
-function addLock(bucket, providerName, filename) {
-  return s3PutObject({
+async function addLock(
+  bucket: string,
+  providerName: string,
+  filename: string
+): Promise<void> {
+  await s3PutObject({
```

Comment:
Any reason for `await` over `return`?

Reply:
The return value from `addLock` was never being used, so I took the opportunity to hide how it was implemented by not returning anything.